tlspuffin/src/openssl/mod.rs

```diff
@@ -77,12 +71,47 @@ pub fn new_openssl_factory() -> Box<dyn Factory<TLSProtocolBehavior>> {
             })?))
         }
 
+        fn kind(&self) -> PutKind {
+            PutKind::Rust
+        }
+
         fn name(&self) -> PutName {
             OPENSSL111_PUT
         }
 
-        fn version(&self) -> String {
-            OpenSSL::version()
+        fn versions(&self) -> Vec<(String, String)> {
+            let openssl_shortname = if cfg!(feature = "openssl101f") {
+                "openssl101f"
+            } else if cfg!(feature = "openssl102u") {
+                "openssl102u"
+            } else if cfg!(feature = "openssl111") {
+                "openssl111k"
+            } else if cfg!(feature = "openssl111j") {
+                "openssl111j"
+            } else if cfg!(feature = "openssl111u") {
+                "openssl111u"
+            } else if cfg!(feature = "openssl312") {
+                "openssl312"
+            } else if cfg!(feature = "libressl") {
+                "libressl333"
+            } else {
+                "unknown"
+            };
```

Comment:
I think it would be nice to get the actual openssl version here.

I think the previous implementation actually called the openssl library.

Reply:
Yes, but the change was done on purpose.

Unlike what we now have in Rust, several C PUTs can be present at the same time in tlspuffin. In next PRs, the plan is to introduce a way to select the PUT through the CLI. We will need a shortname to refer to a specific PUT and that is the purpose of this variable.

Note that the features will disappear when Rust PUTs are removed and we plan to use the name given by the user to `mk_vendor` instead. So there will be a single name that the user has chosen to refer to a specific PUT.